src/ape/managers/project.py

```diff
@@ -59,14 +60,16 @@ def __post_init__(self):
 
         config = self.config.load()
         self.dependencies = {
-            n: self._extract_dependency_manifest(n, dep_id)
-            for n, dep_id in config.dependencies.items()
+            d.name: self._extract_dependency_manifest(d) for d in config.dependencies
         }
 
     def __repr__(self):
         return "<ProjectManager>"
 
-    def _extract_dependency_manifest(self, name: str, download_path: str) -> PackageManifest:
+    def _extract_dependency_manifest(
+        self, dependency_item: Union[IPFSDependency, GithubDependency]
```

Comment:
See? I knew this would come in handy somewhere else!

Maybe our new union type should live in `ape.types` :thinking:

Reply:
Yes, I really do like this typing strategy